### PR TITLE
Compile command

### PR DESCRIPTION
**Pull request by dastanbeksamatov**
#### Steps to test
1. Inside `/cli` run `yarn install && yarn build`
2. Create new global link `npm link` (yarn link does not support global linking, so we use npm)
2. Create a new Subsembly project in a new empty directory with 
     - `subsembly init` or `subsembly init --to=new_sub`
3. Inside the new project run `subsembly compile`

### CODE REVIEW

File: cli/src/commands/commands.ts [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
+        // @ts-ignore
+        builder: (yargs) => {
+            yargs.positional('to', {
+                describe: 'Directory to initialize new Subsembly project in. Defaults to current directory.',
+                type: 'string'
+            })
+        },
+        //@ts-ignore
+        handler: async (argv) => {
+            await Init.run(argv.to ? argv.to : '');
```
Comment (Daniel-K-Ivanov) [inline cli/src/commands/commands.ts]: nit: could be rewritten to:
`Init.run(argv.to || '')`

File: cli/src/commands/commands.ts [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
+        }
     },
-    //@ts-ignore
-    handler: async (argv) => {
-        await Init.run(argv.to ? argv.to : '');
+    {
+        command: 'compile',
+        description: 'Compile Subsembly project',
```
Comment (Daniel-K-Ivanov) [inline cli/src/commands/commands.ts]: We could have the following description:
`Compiles the Subsembly project placed in the current directory to a WASM bytecode`

File: cli/src/commands/compile.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { execSync } from "child_process";
+import fs from "fs";
+import path from 'path';
+import { generateDispatcher, generateFile, generateMetadata } from '../../utils/src';
+import { Constants } from "../constants";
+import { Utils } from "../utils";
+
+export class Compile {
```
Comment (Daniel-K-Ivanov) [inline cli/src/commands/compile.ts:7]: nit: you do not need to have `_` prefixes. You can specify the methods as `private`

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]
<del> I think that once we run `subsembly` the `subsembly help` commands should be shown. </del>
<del> `.github` folder should not be placed inside the newly inited project </del>
<del> I do not think that users need the `metadata.json` file to be generated in their directory </del>
<del> We can remove `tests` folder from the directory as-well </del>
<del> - Do we need the `scripts` folder now or is it going to be later removed after `build-spec`? </del>
<del> We should put the `wasm-code` inside the `build` folder and not in the main directory and we could name it `wasm-runtime` maybe for a better naming? Feel free to suggest another name... </del>

- We shouldn't allow `subsembly init` to be executed in a non-empty directory

File: cli/package.json [hunk 4]
```diff
@@ -28,6 +36,10 @@
     "typescript": "^4.1.3"
   },
   "dependencies": {
+    "@polkadot/keyring": "^5.5.1",
+    "@polkadot/metadata": "^2.10.1",
+    "@polkadot/types": "^2.10.1",
+    "@polkadot/util": "^4.2.1",
```
Comment (Daniel-K-Ivanov) [inline cli/package.json:42]: are we using all those packages?

**Comment by Daniel-K-Ivanov** [comment]
It seems that the Smoke Tests are not going through as-well

File: cli/package.json (hunk 4 above)
Comment (dastanbeksamatov) [reply cli/package.json:42]: yes, `metadata` and `dispatcher` are using them. `@polkadot/keyring` was needed because of typescript types

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]
only the `subsembly init` in non empty dir is not resolved and the CI CD failing, other than that LGTM!

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**